frontend/src/modules/editor/components/Editor.js

```diff
@@ -55,6 +55,10 @@ export class EditorBase extends React.Component<InternalProps, State> {
     constructor(props: InternalProps) {
         super(props);
 
+        // This state acts as a buffer between the Editors and the editor state.
+        // We want to avoid updating the redux state at every key stroke, thus
+        // we keep track of changes in this state and only update the store
+        // every few milliseconds using debouce.
```

Comment:
What's the benefit of updating every few milliseconds vs. updating at every key stroke?

Reply:
Users can type very very fast, and for every key stroke, we would trigger a React render. That can completely boast the UI and make the user experience horrible. I have no idea what the correct time here should be, but I'm sure it's better to have something rather than just let the events roam free.